tests/unit/CMakeLists.txt

```diff
@@ -38,707 +38,707 @@ set(COMMON_COMPILE_OPTIONS
+target_include_directories(ot-test-steering-data
     PRIVATE
         ${COMMON_INCLUDES}
 )
 
-target_compile_options(test-steering-data
+target_compile_options(ot-test-steering-data
     PRIVATE
         ${COMMON_COMPILE_OPTIONS}
 )
 
-target_link_libraries(test-steering-data
+target_link_libraries(ot-test-steering-data
     PRIVATE
         ${COMMON_LIBS}
 )
 
-add_test(NAME test-steering-data COMMAND test-steering-data)
+add_test(NAME ot-test-steering-data COMMAND ot-test-steering-data)
 
-add_executable(test-string
+add_executable(ot-test-string
     test_string.cpp
 )
 
-target_include_directories(test-string
+target_include_directories(ot-test-string
     PRIVATE
         ${COMMON_INCLUDES}
 )
 
-target_compile_options(test-string
+target_compile_options(ot-test-string
     PRIVATE
         ${COMMON_COMPILE_OPTIONS}
 )
 
-target_link_libraries(test-string
+target_link_libraries(ot-test-string
     PRIVATE
         ${COMMON_LIBS}
 )
 
-add_test(NAME test-string COMMAND test-string)
+add_test(NAME ot-test-string COMMAND ot-test-string)
 
-add_executable(test-timer
+add_executable(ot-test-timer
     test_timer.cpp
 )
 
-target_include_directories(test-timer
+target_include_directories(ot-test-timer
     PRIVATE
         ${COMMON_INCLUDES}
 )
 
-target_compile_options(test-timer
+target_compile_options(ot-test-timer
     PRIVATE
         ${COMMON_COMPILE_OPTIONS}
 )
 
-target_link_libraries(test-timer
+target_link_libraries(ot-test-timer
     PRIVATE
         ${COMMON_LIBS}
 )
 
-add_test(NAME test-timer COMMAND test-timer)
+add_test(NAME ot-test-timer COMMAND ot-test-timer)
 
 set_target_properties(
-    test-platform
-    test-aes
-    test-checksum
-    test-child
-    test-child-table
-    test-cmd-line-parser
-    test-dns
-    test-ecdsa
-    test-flash
-    test-heap
-    test-hkdf-sha256
-    test-hmac-sha256
-    test-ip-address
-    test-link-quality
-    test-linked-list
-    test-lookup-table
-    test-lowpan
-    test-mac-frame
-    test-macros
-    test-message
-    test-message-queue
-    test-multicast-listeners-table
-    test-ndproxy-table
-    test-netif
-    test-network-data
-    test-pool
-    test-priority-queue
-    test-pskc
-    test-steering-data
-    test-string
-    test-timer
+    ot-test-platform
+    ot-test-aes
+    ot-test-checksum
+    ot-test-child
+    ot-test-child-table
+    ot-test-cmd-line-parser
+    ot-test-dns
+    ot-test-ecdsa
+    ot-test-flash
+    ot-test-heap
+    ot-test-hkdf-sha256
+    ot-test-hmac-sha256
+    ot-test-ip-address
+    ot-test-link-quality
+    ot-test-linked-list
+    ot-test-lookup-table
+    ot-test-lowpan
+    ot-test-mac-frame
+    ot-test-macros
+    ot-test-message
+    ot-test-message-queue
+    ot-test-multicast-listeners-table
+    ot-test-ndproxy-table
+    ot-test-netif
+    ot-test-network-data
+    ot-test-pool
+    ot-test-priority-queue
+    ot-test-pskc
+    ot-test-steering-data
+    ot-test-string
+    ot-test-timer
```

Comment:
Looks good to me. 

Can we also consider changing the `Makefile.am` so the test names remain the same?
https://github.com/openthread/openthread/blob/66802336dec9b8ba4850b054615fe79bf8a01c4a/tests/unit/Makefile.am#L108-L116